guides/satellite-sync.sh

```diff
@@ -3,14 +3,49 @@
 # ##################################################
 # Created on: Fri, 23 Oct 2019
 # Created by: Andrew Dahms
-# Hacked repeately by Melanie Corr to accomodate transitions towards Upstream-Downstream Satellite
+# Hacked repeately by Melanie Corr to accommodate transitions towards Upstream-Downstream Satellite
+# To use the script to synchronize changes from the upstream foreman-documentation GitHub repository to the downstream docs-Red_Hat_Satellite_6 repository, complete the following steps:"
+# 1. Create a directory to store repositories for synchronization. You can name the directory whatever you want, for example SyncCenter:"
+# $ mkdir ~/Documents/SyncCenter"
+#
+# 2. Copy this script to the SyncCenter directory."
+#
+# 3. Change to the SyncCenter directory."
+#
+# 4. In the upstream repository, ensure that the commits that you want to synchronized are cherry-picked to the related branch, for example, to the SATELLITE-6.8-beta branch. In the upstream repository, all related to downstream branches are in the format 'SATELLITE-x.y'"
+#
+# 5. Run the script and specify the upstream branch to copy changes from and the downstream branch to copy changes to. For example, to synchronize between the upstream SATELLITE-6.8-beta branch and the downstream master branch, enter the following command:"
+# $ sh satellite-sync.sh SATELLITE-6.8-beta master"
+#
+# This command copies the changes to the downstream repo and commits them."
+# Note that you can run the script with the '--noop' option to copy changes but not commit them."
 # ##################################################
 
 echo ========================================
 echo Satellite Upstream-Downstream Synchronizer
 echo ========================================
 echo
 
+#Print help.
+if [ "$1"  == "?" ] || [ "$1"  == "-?" ] || [ "$1"  == "--help" ] || [ "$1"  == "-h" ]; then
+  echo "To use the script to synchronize changes from the upstream foreman-documentation GitHub repository to the downstream docs-Red_Hat_Satellite_6 repository, complete the following steps:"
+  echo "1. Create a directory to store repositories for synchronization. You can name the directory whatever you want, for example SyncCenter:"
+  echo "$ mkdir ~/Documents/SyncCenter"
+  echo ""
+  echo "2. Copy this script to the SyncCenter directory."
+  echo ""
+  echo "3. Change to the SyncCenter directory."
+  echo ""
+  echo "4. In the upstream repository, ensure that the commits that you want to synchronized are cherry-picked to the related branch, for example, to the SATELLITE-6.8-beta branch. In the upstream repository, all related to downstream branches are in the format 'SATELLITE-x.y'"
+  echo ""
+  echo "5. Run the script and specify the upstream branch to copy changes from and the downstream branch to copy changes to. For example, to synchronize between the upstream SATELLITE-6.8-beta branch and the downstream master branch, enter the following command:"
```

Comment:
I understand this is just an example. But in reality, do we commonly sync between the upstream SATELLITE-6.8-beta branch and the downstream master branch, or the downstream 6.8-beta branch? I was unsure myself so I didn't try out the syncing step. If it's the latter, I would suggest updating the text and example to reflect equivalent beta branches to avoid confusion.

Reply:
The master branch is the branch that is supposed to have all the latest changes. Every commit that belong to the very recent Satellite version, in our case it is unreleased 6.8-beta, should be cherry-picked to master first. So you always need to synchronize changes to master anyway.
The overall structure of repositories look like this:
Upstream `master` is in sync with downstream `master`
Upstream `SATELLITE-6.7` is in sync with downstream `6.7`
Upstream `SATELLITE-6.8-beta` is in sync with downstream `6.8-beta`
In general, upstream `SATELLITE-x.y` is in sync with downstream `x.y`.

That is the state that we need to maintain. Here is one method how you can maintain it:

1. Upstream, create branch with your work from `master`, push changes, create a PR and have it merged. Note that if you need to change something for an older version ONLY, for example if the change is related to `6.7` but should not be reflected on `master`, you push this commit to `SATELLITE-6.7` directly, though this is not a common scenario.

2. After the PR is merged, cherry-pick the changes to the relevant upstream branches.

- The change can be only relevant to the `6.8-beta` version of Satellite, for examle the Foreman Ansible Modulse PR. In this case, only cherry-pick it to `SATELLITE-6.8-beta`. 
- The change can also be relevant to more versions, for example to `6.7` and `6.8-beta`. In this case, cherry-pick the change to all relevant SATELLITE branches.

3. When the changes are in the relevant branches, you can use the `satellite-sync.sh` script to copy them to downstream branches. You can do this by running a command for each branch, for example, for Foreman Ansible modules PR, you will need to run those two commands:
`sh satellite-sync.sh master master`
`sh satellite-sync.sh SATELLITE-6.8-beta 6.8-beta`

That's it, I will try to implement this somehow to the satellite-sync readme.